src/widget/wpushbutton.cpp

```diff
@@ -348,12 +348,12 @@ void WPushButton::mousePressEvent(QMouseEvent * e) {
     if (m_leftButtonMode == ControlPushButton::POWERWINDOW
             && m_iNoStates == 2) {
         if (leftClick) {
-            if (getControlParameterLeft() == 0.0) {
-                m_clickTimer.setSingleShot(true);
-                m_clickTimer.start(ControlPushButtonBehavior::kPowerWindowTimeMillis);
-            }
+            m_clickTimer.setSingleShot(true);
+            m_clickTimer.start(ControlPushButtonBehavior::kPowerWindowTimeMillis);
+
+            double value = getControlParameterLeft() == 0.0 ? 1.0 : 0.0;
+            setControlParameterLeftDown(value);
```

Comment:
These two lines could be combined into a (reusable) function: toggleControlParameterLeft()

Reply:
this function is already quite compact. isolating this into a separate function would generate more lines of code than necessary because no other function in wpushbutton is using `setControlParameterLeftDown`.
adding such a toggle function to wbasewidget would also create more complexity considering we'd have to add polling there, also no other widget except wpushbutton is using `setControlParameterLeftUp`.
that said I'd rather leave as it is.

Reply:
would a 'toggle' comment do fo you?